presto-main/src/main/java/com/facebook/presto/memory/NodeMemoryConfig.java

```diff
@@ -55,4 +58,18 @@ public NodeMemoryConfig setMaxQueryTotalMemoryPerNode(DataSize maxQueryTotalMemo
         this.maxQueryTotalMemoryPerNode = maxQueryTotalMemoryPerNode;
         return this;
     }
+
+    public double getHeapHeadroom()
+    {
+        return heapHeadroom;
+    }
+
+    @Config("memory.heap-headroom-per-node")
+    @ConfigDescription("The fraction of heap to set aside as headroom/buffer (e.g., for untracked allocations)")
+    public NodeMemoryConfig setHeapHeadroom(double heapHeadroom)
+    {
+        checkArgument(Double.compare(heapHeadroom, 0.0) >= 0 && Double.compare(heapHeadroom, 1.0) < 0, "heapHeadroom must be in the [0, 1) range");
```

Comment:
Ask @dain about double constraint.

`Double.compare(heapHeadroom, 0.0) >= 0` -> `heapHeadroom >= 0.0`?

Reply:
`Double.compare` is useful when you care about `heapHeadroom` being NaN. However, with `heapHeadroom` being `NaN`, `heapHeadroom >= 0` (and also `0 <= heapHeadroom`) would be false. So this can just be 
```
checkArgument(0.0 <= heapHeadroom && heapHeadroom < 1.0, "heapHeadroom must be in the [0, 1) range");
```

Btw, why you do not allow `heapHeadroom == 1` ? I suppose `heapHeadroom == 0.99999999999` is as bad (or as good) as `1`?

Reply:
I have changed the lower/upper bounds.